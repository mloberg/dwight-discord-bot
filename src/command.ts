import { SlashCommandBuilder, SlashCommandIntegerOption } from '@discordjs/builders';
import { CommandInteraction } from 'discord.js';

type Handler = (command: CommandInteraction) => Promise<void>;

export class CommandBuilder extends SlashCommandBuilder {
    constructor(readonly handler: Handler) {
        super();
    }
}

export const rollOption = (dice: string): SlashCommandIntegerOption =>
    new SlashCommandIntegerOption().setName('roll').setDescription(`Value of a ${dice} roll.`);
