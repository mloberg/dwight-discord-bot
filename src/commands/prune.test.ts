import { Client, Guild, Message, TextChannel } from 'discord.js';

import { FriendlyError } from '../error';
import command from './prune';

const mocks = {
    delete: jest.fn(),
};

jest.mock('discord.js', () => ({
    Client: jest.fn(),
    Guild: jest.fn(),
    TextChannel: jest.fn(),
    Collection: jest.fn(),
    Message: jest.fn().mockImplementation(() => {
        return {
            channel: {
                type: 'text',
                bulkDelete: mocks.delete,
            },
        };
    }),
}));

describe('_prune configuration', () => {
    it('should have basic command infomation', () => {
        expect(command.name).toEqual('prune');
        expect(command.description).toEqual('Prune messages from a channel');
        expect(command.usage).toEqual('COUNT');
    });

    it('should have a purge alias', () => {
        expect(command.alias).toContain('purge');
    });
});

describe('_ping', () => {
    let message: Message;

    beforeEach(() => {
        mocks.delete.mockClear();

        const client = new Client();
        const guild = new Guild(client, {});
        const channel = new TextChannel(guild, {});
        message = new Message(client, {}, channel);
    });

    it('deletes messages in text channels', async () => {
        await command.run(message, { $0: 'prune', _: ['3'] });

        expect(mocks.delete).toBeCalledWith(4, true);
    });

    it('throws an error if in a DM', async () => {
        message.channel.type = 'dm';

        try {
            await command.run(message, { $0: 'prune', _: ['3'] });

            fail('expected error to be thrown');
        } catch (err) {
            expect(err instanceof FriendlyError).toBe(true);
            expect(err.message).toEqual("I can't bulk delete DMs.");
        }
    });
});
