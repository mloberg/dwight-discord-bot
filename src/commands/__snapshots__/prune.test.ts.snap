// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/prune is a slash command 1`] = `
Object {
  "description": "Prune messages",
  "name": "prune",
  "options": Array [
    Object {
      "choices": undefined,
      "description": "Number of messages to delete",
      "name": "messages",
      "required": true,
      "type": 4,
    },
  ],
}
`;
